### PR TITLE
patches final reorg

### PR DESCRIPTION
**Pull request by ThyWoof**
. final reorg step;
. moved level 20 feature related patches into Custom Features;
. moved remaining Power patches into Custom Features;
. created new CustomFeaturesContext to hold any shared code;

### CODE REVIEW

File: SolastaCommunityExpansion/Models/CustomFeaturesContext.cs [hunk 1]
```diff
@@ -158,5 +87,25 @@ private static int GetMaxUsesForPool(RulesetUsablePower poolPower, RulesetCharac
...
+        internal static void UpdateUsageForPowerPool(this RulesetCharacter character, RulesetUsablePower modifiedPower, int poolUsage)
+        {
+            if (!(modifiedPower.PowerDefinition is IPowerSharedPool))
+            {
+                return;
+            }
+            IPowerSharedPool sharedPoolPower = (IPowerSharedPool)modifiedPower.PowerDefinition;
+            foreach (RulesetUsablePower poolPower in character.UsablePowers)
+            {
+                if (poolPower.PowerDefinition == sharedPoolPower.GetUsagePoolPower())
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Models/CustomFeaturesContext.cs:100]: sharedPoolPower.GetUsagePoolPower() - move outside loop

Comment (ThyWoof) [reply SolastaCommunityExpansion/Models/CustomFeaturesContext.cs:100]: Will do. I wasn’t willing to touch any logic here but good optimization catch.

**Review by PhilPJL — CHANGES_REQUESTED** [review]
See comment

File: SolastaCommunityExpansion/Patches/CustomFeatures/RulesetCharacterPatcher.cs [hunk 2]
```diff
@@ -46,4 +46,153 @@ internal static void Postfix(RulesetCharacter __instance)
...
+            // Set the minimum die roll based on the bonuses, which indirectly sets the minimum total.
+            // This can result in impossible die rolls, like rolling 23 on a d20, which the game doesn't seem to mind.
+            return minimumTotal.Value - (baseBonus + rollModifier);
+        }
+
+        private static int? MinimumStrengthAbilityCheckTotal(RulesetCharacter character, string proficiencyName)
+        {
+            var featuresToBrowse = new List<FeatureDefinition>();
+
+            character.EnumerateFeaturesToBrowse<IMinimumAbilityCheckTotal>(featuresToBrowse);
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Patches/CustomFeatures/RulesetCharacterPatcher.cs:139]: just a comment but when using this pattern it's safe to use the helper

`var featuresToBrowse = character.EnumerateFeaturesToBrowse<IMinimumAbilityCheckTotal>(featuresToBrowse);`

and then you don't need the OfType<>

**Review by PhilPJL — APPROVED** [review]

**Pull request merged**